sw/otbn/code-snippets/randomness.s

```diff
@@ -72,14 +78,7 @@ test_rnd:
   /* Read the RND WSR again. */
   bn.wsrr w1, 0x1 /* RND */
 
-  /* Fail the test if the two numbers read from RND are equal. */
-  bn.cmp w0, w1
-  csrrs x2, 0x7c0 /* FG0 */, x0
-  andi x2, x2, 8 /* filter out Z (zero) bit */
-  addi x31, x0, 3
-  bne x2, x0, exit_fail
-
-  ret
+  jal x0, error_checking
```

Comment:
This subroutine should still end in a `ret`:
```suggestion
  jal x0, error_checking
  ret
```

Reply:
To use a `ret` here I would have to stack the `x1` before calling the `error_checking` subroutine. Is that what you mean, or just add it to delimit the end of the subroutine?

Reply:
Ah, I missed that it was `x0` here! In that case this does make sense, although it's a bit more idiomatic in OTBN to use `jal x1, error_checking` and then a `ret`. Up to you, though.

Reply:
I see. Well to use `jal x1, error_checking` followed by a `ret` I would also need to add two extra instructions to save the `x1` and restore it. At the end we would have 6 extra instructions (3 each test). So I think I`ll leave as it is if you don't mind.
Thanks.

Reply:
@doreis-lowrisc: Jade's right here (our `x1` register is a magic stack). But I'm also fine with tail calls, so don't mind leaving it as is.

Reply:
ohh, I didn`t know that OTBN had this feature. In that case I'll change it.
Thanks!